### PR TITLE
vcpkg build support

### PR DESCRIPTION
**Pull request by sergey-chernikov**
Debug build on Linux is triggered by `-DCMAKE_BUILD_TYPE=Debug` at cmake run
Tests are toggled by `-DENABLE_TESTS=ON/OFF` (on by default)

### CODE REVIEW

File: cmake/FindPython2.cmake [hunk 1]
```diff
@@ -0,0 +1,171 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
```
Comment (goatpig) [inline cmake/FindPython2.cmake:1]: py2 requirement has been removed from GUI, is there a need to reintroduce it for cmake? Cant it used py3?

File: cppForSwig/Signer/ResolverFeed.h [hunk 2]
```diff
@@ -17,7 +17,7 @@
 #include "../SecureBinaryData.h"
 #include "../Wallets/BIP32_Node.h"
 
-#include "../protobuf/Signer.pb.h"
+#include "protobuf/Signer.pb.h"
```
Comment (goatpig) [inline cppForSwig/Signer/ResolverFeed.h:20]: Is this change necessary? It may break the autotools build

File: cppForSwig/Signer/ResolverFeed_Wallets.cpp [hunk 3]
```diff
@@ -20,7 +20,7 @@ using namespace Armory::Wallets;
 //// ResolverFeed_AssetWalletSingle
 ////
 ////////////////////////////////////////////////////////////////////////////////
-void ResolverFeed_AssetWalletSingle::addToMap(shared_ptr<AddressEntry> addrPtr)
+void Armory::Signer::ResolverFeed_AssetWalletSingle::addToMap(shared_ptr<AddressEntry> addrPtr)
```
Comment (goatpig) [inline cppForSwig/Signer/ResolverFeed_Wallets.cpp:23]: This change is way too disruptive, the cpp file should use namesapce Armory::Signer as a one liner alternative

File: cppForSwig/Signer/Transactions.cpp [hunk 4]
```diff
@@ -21,7 +21,7 @@ TransactionStub::~TransactionStub(void)
 //// TransactionVerifier
 ////////////////////////////////////////////////////////////////////////////////
 ////////////////////////////////////////////////////////////////////////////////
-bool TransactionVerifier::verify(bool noCatch, bool strict) const
+bool Armory::Signer::TransactionVerifier::verify(bool noCatch, bool strict) const
```
Comment (goatpig) [inline cppForSwig/Signer/Transactions.cpp:24]: use namespace Armory::Signer as a one liner solution

File: cppForSwig/libbtc/src/secp256k1/src/tests.c [hunk 5]
```diff
@@ -1,4525 +0,0 @@
-/**********************************************************************
```
Comment (goatpig) [inline cppForSwig/libbtc/src/secp256k1/src/tests.c:1]: Is it necessary to delete a test suite?

File: cmake/FindPython2.cmake (hunk 1 above)
Comment (sergey-chernikov) [reply cmake/FindPython2.cmake]: Removed, and disabled Qt4 and py2 from cmake building - I guess it will be superceded by py3 and Qt5 soon anyway.

File: cppForSwig/Signer/Transactions.cpp (hunk 4 above)
Comment (sergey-chernikov) [reply cppForSwig/Signer/Transactions.cpp:24]: no, VS complains about ambiguous names if not added explicitly

File: cppForSwig/libbtc/src/secp256k1/src/tests.c (hunk 5 above)
Comment (sergey-chernikov) [reply cppForSwig/libbtc/src/secp256k1/src/tests.c:1]: I actually only copied libbtc subdir from the upstream goapig/BitcoinArmory dev branch, so it's likely deleted there (I don't see it in my cloned copy)

File: cppForSwig/Signer/ResolverFeed_Wallets.cpp (hunk 3 above)
Comment (sergey-chernikov) [reply cppForSwig/Signer/ResolverFeed_Wallets.cpp:23]: See response for Transactions.cpp - for some reason full namespace is required to build these classes. Otherwise it's a compile error (not warning).

File: cppForSwig/Signer/ResolverFeed.h (hunk 2 above)
Comment (sergey-chernikov) [reply cppForSwig/Signer/ResolverFeed.h:20]: Yes, otherwise it can't find the Signer.pb.h. Overall it's a good idea to get rid of all relative paths (`../`) for includes - I'd better fix the autotools builds (by adding appropriate include dirs) once approach the libraries separation PRs.

**Comment by goatpig** [comment]
An explaination of what this PR is about:
- Blocksettle uses Armory as is for its Bitcoin handling. They have a cmake project to build the Armory code.
- I used to develop Armory on Windows, but since then have permanently moved to Linux.
- Armory uses autotool for its Linux build system, this doesn't translate well to Windows.
- For Windows, it uses a set of handcrafted .vcproj and .sln msvc files. These are old and I want to get rid of them at any rate.
- cmake generates msvc projects/solutions on Windows

Hence, Sergey (from Blocksettle) is contributing their cmake project back to Armory. The end goal is to have a single go to script for users to setup and build Armory from source on Windows & Linux (maybe even Mac). I'll keep on using autotools and setting things up manually when I dev.
